auth/server/src/main/java/com/seagates3/authorization/Authorizer.java

```diff
@@ -79,7 +79,47 @@ ServerResponse authorize(Requestor requestor,
     ServerResponse serverResponse = null;
     AuthorizationResponseGenerator responseGenerator =
         new AuthorizationResponseGenerator();
-    String mainOperation = requestBody.get("S3Action");
+    String s3Operation = requestBody.get("S3Action");
+    if (AuthServerConfig.isEnableIamPolicy()) {
+      // Authorize IAM Policy
+      serverResponse =
+          new IAMPolicyAuthorizer().authorizePolicy(requestor, requestBody);
+    }
+    /* If s3Operation is NULL means call is for IAM operation. eg-
+     * ListAccessKeys so return
+     *     whatever is iam policy authorization response. Dont go for Bucket
+     * policy authorization.
+     * If s3Operation is not null means its s3 operation authorization call.
+     * so check first iam policy-
+     *     if iam policy response is not null and not ok then return
+     * AccessDenied.Dont check bucket policy auth.
+     *     if iam policy authorization response is either null or ok then check
+     * bucket policy auth
+     */
+    if (s3Operation == null && serverResponse == null) {
+      try {
+        if (RootPermissionAuthorizer.getInstance().containsAction(
+                requestBody.get("Action"))) {
+          new IAMApiAuthorizer().authorizeRootUser(requestor, requestBody);
+        } else {
+          new IAMApiAuthorizer().authorize(requestor, requestBody);
+        }
+        LOGGER.info("Request is authorized for user: " + requestor.getName() +
```

Comment:
Grammer for logger statment is wrong. please rephrase sentence

Reply:
Fixed